src/app/customizable-demo-form.component.html

```diff
@@ -10,31 +10,36 @@
        [step]="1"
        [disabled]="customizeScoreThresholds"
        [(ngModel)]="sliderValue"
-       (change)="onSliderValueChange($event)"></md-slider>
+       (change)="onSliderValueChange($event); onDemoSettingsChanged()"></md-slider>
```

Comment:
(this applies for the rest of the file):

Having the template deal with the multple calls feels a bit odd to me and like it should be in the controller, but I'm not familiar with Angular style.  Is there a reason to do it this way?

Reply:
Not sure what you mean by the controller. I thought this sort of thing (event-driven code) was an Angular pattern; we see a change in our child, we call a function (which updates the URL, for this case). I could pull the onDemoSettingsChanged() call into each of the the other callbacks (like onSliderValueChange) but I think it's clearer to see what's going on when it's in the template like this.

Before, what I had was not event driven at all; the data binding would just call getDemoSettings() a bunch of times per second (every change detection cycle), and update the child continuously (I originally had it that way because sadly angular doesn't listen on changes to child properties of an object, but the object reference itself). Now it's less expensive because it doesn't have to call getDemoSettings every cycle, but only when we see a child element change.

But if you have another suggestion, I'm definitely open to hearing it.